recursive-feature-elimination/script.whizzml

```diff
@@ -0,0 +1,248 @@
+  ;; choosable-objective-ids
+  ;;
+  ;; List of ids of the fields in the dataset that can be chosen as objective
+  ;; field.
+  ;;
+  ;; Inputs:
+  ;;  fields: (map) Fields structure
+  ;; Output: (list) list of field ids
+  (define (choosable-objective-ids fields)
+    (let (field-val (lambda (fid k) (fields [fid k] false))
+          objective-types ["categorical", "numeric"]
+          pref? (lambda (k) (field-val k "preferred"))
+          pred? (lambda (k) (member? (field-val k "optype") objective-types)))
+      (filter (lambda (x) (and (pref? x) (pred? x))) (keys fields))))
+
+
+  ;; safe-delete
+  ;;
+  ;; deletes resource ignoring errors
+  ;; Inputs:
+  ;;   id: (resource-id) resource to be deleted
+  ;;
+  ;; Output: (boolean) true if successful, false if not
+  ;;
+  (define (safe-delete id)
+    (try (delete id)
+         (catch e
+           (log-info (str "Error deleting resource " id " ignored"))
+           false)))
+
+  ;; default-inputs
+  ;;
+  ;; Get the default set of input fields for this dataset (all preferred
+  ;; fields minus the objective field).
+  ;;
+  ;; Inputs:
+  ;;  dataset-id: (dataset-id) Dataset id
+  ;;  obj-id: (string) id of the objective field
+  ;;
+  ;; Output: (list) Inputs field ids for this dataset
+  (define (default-inputs dataset-id obj-id)
+    (let (fields ((fetch dataset-id) "fields")
+          fids (keys fields))
+      (filter (lambda (k) (and (fields [k "preferred"] true) (not (= obj-id k))))
+              fids)))
+
+  ;; random-forest
+  ;; Create a random forest from the dataset
+  ;;
+  ;; Inputs:
+  ;;   ds-id: (dataset-id) Dataset id
+  ;;   obj: (string) Objective field id
+  ;;   run: (integer) Number of run (to change the seed)
+  ;;   input-fields: (list) Specifies the fields to be considered to create the ensemble.
+  ;; Output: (ensemble-id) Random Forest id
+  (define (random-forest ds-id obj run, input-fields)
+    (create-and-wait-ensemble {"dataset" ds-id
+                               "objective_field" obj
+                               "randomize" true
+                               "random_candidate_ratio" 0.33
+                               "sample_rate" 0.8
+                               "seed" (str "BigML Boruta" run)
+                               "input_fields" input-fields
+                               "ensemble_sample" {"rate" 1
+                                                  "replacement" true
+                                                  "seed" (str "BigML Boruta" 
+                                                               run) }}))
+
+;; evaluate-model
+;;
+;; Creates and returns an evaluation for a model
+;;
+;; Inputs:
+;;   model-id: (model-id) Model to evaluate
+;;   test-ds-id: (dataset-id) Test Dataset id
+;; Output:
+;;   evaluation: (evaluation) Result of the model evaluation
+
+  (define (evaluate-model model test-ds-id)
+    (create-and-wait-evaluation {"dataset" test-ds-id
+                                "model" model}))
```

Comment:
this function is so trivial that i wouldn't even define it... i'd just call:

```
(wait (create-evaluation test-ds-id model))
```
wherever you're calling `evaluate-model`.  note that it's okay to just use the model and dataset ids directly (and order doesn't matter).